### PR TITLE
chore: Rust  toolchain update

### PR DESCRIPTION
**Pull request by dpc**
I was looking into some slowness in the CI, and it would be improved with sparse index support which was recently made stable

https://blog.rust-lang.org/2023/03/09/Rust-1.68.0.html#cargos-sparse-protocol

(+ made default in 1.70).

There's a separate PR for miniscript update, which seems to be required here as well: https://github.com/fedimint/fedimint/pull/2950

### CODE REVIEW

**Review by justinmoon — APPROVED** [review]

File: flake.nix [hunk 1]
```diff
@@ -28,12 +28,18 @@
...
           };
 
-          pkgs-unstable = import nixpkgs-unstable {
+          pkgs = import nixpkgs {
             inherit system;
+            overlays = [
+              (final: prev: {
+                cargo-udeps = pkgs-unstable.cargo-udeps;
+              })
+            ];
```
Comment (elsirion) [inline flake.nix:42]: What's the benefit of using the unstable udeps version this way instead of directly?

File: devimint/src/lib.rs [hunk 2]
```diff
@@ -141,7 +141,7 @@ impl Gatewayd {
 
     pub async fn connect_fed(&self, fed: &Federation) -> Result<()> {
         let invite_code = fed.invite_code()?;
-        poll_max_retries("gateway connect-fed", 10, || async {
+        poll_max_retries("gateway connect-fed", 60, || async {
```
Comment (elsirion) [inline devimint/src/lib.rs:144]: Is this related to #2971?

**Review by elsirion — APPROVED** [review]

File: flake.nix (hunk 1 above)
Comment (dpc) [reply flake.nix:42]: Can't forget about using the right once. :D

File: devimint/src/lib.rs (hunk 2 above)
Comment (dpc) [reply devimint/src/lib.rs]: Yes. Removing it as it was moved to another PR.
